charts/nginx-ingress/values.yaml

```diff
@@ -167,6 +167,26 @@ controller:
   #   cpu: 1
   #   memory: 1Gi
 
+  ## The security context for the Ingress Controller pods.
+  podSecurityContext:
+    seccompProfile:
+      type: RuntimeDefault
+
+  ## The security context for the Ingress Controller containers.
+  securityContext: {} # Remove curly brackets before adding values
```

Comment:
I think we need to keep `securityContext` commented out otherwise we'll always evaluate to `true` on the `if .Values.controller.securityContext` check

Reply:
It's not actually, seems it actually checks if json is empty or not. Other attributes like affinity and tolerations also follow same pattern.

tested by changing:
`{{- if or .Values.controller.readOnlyRootFilesystem (and .Values.controller.securityContext .Values.controller.securityContext.readOnlyRootFilesystem) -}}`
to 
`{{- if or .Values.controller.readOnlyRootFilesystem .Values.controller.securityContext -}}`

and readOnlyfs related containers weren't created at start.